driver-core/src/main/com/mongodb/operation/BulkWriteBatch.java

```diff
@@ -163,6 +150,21 @@ private BulkWriteBatch(final MongoNamespace namespace, final ConnectionDescripti
         this.unprocessed = unprocessedItems;
         this.payload = new SplittablePayload(getPayloadType(batchType), payloadItems);
         this.sessionContext = sessionContext;
+        this.command = new BsonDocument();
+
+        if (!payloadItems.isEmpty()) {
```

Comment:
Moved this block and added conditional to avoid advancing the transaction number if there is nothing left to send.  That's not required strictly for correctness, but it is required to get the transactions integration tests to pass.